x/payments/spec/03_messages.md

```diff
@@ -0,0 +1,145 @@
+	DiscountId        sdk.Uint       `json:"discount_id" yaml:"discount_id"`
+}
+```
+This message is expected to fail if:
+- any editable field violates the restrictions set for the same field in `MsgCreatePaymentContract`
+- all editable fields are `"[do-not-modify]"`
+
+## MsgCreateSubscription 
+
+The owner of a bond can edit some of the bond's parameters using `MsgCreateSubscription`.
+
+| **Field**              | **Type**           | **Description**                                                                                               |
+|:-----------------------|:-------------------|:--------------------------------------------------------------------------------------------------------------|
+| CreatorDid             | `did.Did`          | Did of the user 
+| SubscriptionId         | `string`           | ID for the subscription
+| PaymentContractId      | `string`           | ID for the paymentContract
+| MaxPeriods             | `sdk.AccAddress`   | 
+| Period                 | `bool`             | IF the periods are allowed or not
+
+
+```go
+type MsgCreateSubscription struct {
+	CreatorDid        did.Did  `json:"creator_did" yaml:"creator_did"`
+	SubscriptionId    string   `json:"subscription_id" yaml:"subscription_id"`
+	PaymentContractId string   `json:"payment_contract_id" yaml:"payment_contract_id"`
+	MaxPeriods        sdk.Uint `json:"max_periods" yaml:"max_periods"`
+	Period            Period   `json:"period" yaml:"period"`
+}
+```
+This message is expected to fail if:
+- any editable field violates the restrictions set for the same field in `MsgCreateSubscription`
+- all editable fields are `"[do-not-modify]"`
+
+## MsgSetPaymentContractAuthorisation 
+
+The owner of a bond can edit some of the bond's parameters using `MsgSetPaymentContractAuthorisation`.
+
+| **Field**              | **Type**           | **Description**                                                                                               |
+|:-----------------------|:-------------------|:--------------------------------------------------------------------------------------------------------------|
+| PayerDid               | `did.Did`          | Payer's DID ID 
+| PaymentContractId      | `string`           | ID of the paymentContract
+| Authorised             | `bool`             | Status of authorisation
+
+
+```go
+type MsgSetPaymentContractAuthorisation struct {
+	PayerDid          did.Did `json:"payer_did" yaml:"payer_did"`
+	PaymentContractId string  `json:"payment_contract_id" yaml:"payment_contract_id"`
+	Authorised        bool    `json:"authorised" yaml:"authorised"`
+}
+```
+This message is expected to fail if:
+- any editable field violates the restrictions set for the same field in `MsgSetPaymentContractAuthorisation`
+- all editable fields are `"[do-not-modify]"`
+
+## MsgGrantDiscount 
+
+The owner of a bond can edit some of the bond's parameters using `MsgGrantDiscount`.
```

Comment:
This mentions bonds but we're not dealing with bonds here.

Reply:
removed